### PR TITLE
Resample to input

### PR DESCRIPTION
**Pull request by yw7**
Introduced '--iso' flag to toggle between isotropic output and resampling to input space. This enhancement allows users to choose whether to maintain the original resolution of the model output or resample it to match input images. Improves flexibility in output handling based on user preference.

### CODE REVIEW

File: totalspineseg/inference.py [hunk 1]
```diff
@@ -698,5 +704,60 @@ def main():
...
+            max_workers=max_workers,
+            quiet=quiet,
+        )
+        if not quiet: print('\n' 'Resampling step1_cord to the input images space:')
+        transform_seg2image_mp(
+            input_path,
+            output_path / 'step1_cord',
+            output_path / 'step1_cord',
+            image_suffix = '',
+            interpolation = 'linear',
```
Comment (NathanMolinier) [inline totalspineseg/inference.py:724]: Shouldn't we use nearest neighbour ? To keep integer values ?

File: totalspineseg/inference.py [hunk 2]
```diff
@@ -698,5 +704,60 @@ def main():
...
+            max_workers=max_workers,
+            quiet=quiet,
+        )
+        if not quiet: print('\n' 'Resampling step1_canal to the input images space:')
+        transform_seg2image_mp(
+            input_path,
+            output_path / 'step1_canal',
+            output_path / 'step1_canal',
+            image_suffix = '',
+            interpolation = 'linear',
```
Comment (NathanMolinier) [inline totalspineseg/inference.py:735]: Same

File: totalspineseg/inference.py (hunk 1 above)
Comment (yw7) [reply totalspineseg/inference.py:724]: No, because this is soft segmentation

File: totalspineseg/inference.py (hunk 2 above)
Comment (yw7) [reply totalspineseg/inference.py:735]: Same (:

File: totalspineseg/inference.py [hunk 3]
```diff
@@ -698,5 +704,60 @@ def main():
             },
         )
 
+    if not iso:
```
Comment (NathanMolinier) [inline totalspineseg/inference.py:707]: We need to deal with the case when only the step1 is run

Comment (yw7) [reply totalspineseg/inference.py:707]: Yes
I think if the folder doesn't exist it just do nothing. But ill test it

Comment (yw7) [reply totalspineseg/inference.py:707]: I'll add an `if` for it as it is not mice to see something like this (: 
0it [00:00, ?it/s]

**Review by NathanMolinier — APPROVED** [review]
Seems to work fine ! Ready to merge

**Pull request merged**